packages/api/src/schema/api-schema.yaml

```diff
@@ -2854,7 +2856,137 @@ components:
         targetSegmentSizeSecs:
           $ref: >-
             #/components/schemas/new-asset-payload/properties/targetSegmentSizeSecs
-
+    # AI Generate payloads. Keep in mind that these use snake_case instead of camelCase since
+    # they implement the same interface as the AI Gateway Livepeer node.
+    text-to-image-payload:
+      type: object
+      required:
+        - prompt
+      properties:
+        prompt:
+          type: string
+        model_id:
+          type: string
+          default: SG161222/RealVisXL_V4.0_Lightning
+          enum:
+            - SG161222/RealVisXL_V4.0_Lightning
```

Comment:
Do I understand correctly that currently, we accept only 2 models and that these 2 specific models are always available in Os? 

Btw. what happens if the model is not available in the specific O?

Reply:
@leszko Forgot to reply this. Yeah right now we accept only these 2 models (or others in other APIs) which are the ones recommended by the AI Network team on the docs. e.g. https://docs.livepeer.org/ai/pipelines/text-to-image#models

Other models (called "on-demand") are still supported, but they might have less support from Orchestrators so I opted to keep them out on this first version. 

Currently, an O can keep only 1 model (from any pipeline) warmed up, as it has to keep an `ai-worker` container running with that model loaded in the GPU. When the model is not available (warm) in the O, it will just kill that running container and start another `ai-worker` configured with the requested model. The model is usually available on the disk if the operator previously configured it (see [this getting started](https://docs.livepeer.org/ai/orchestrators/models-download)), otherwise the `ai-worker` will download the model dynamically from [hugging face](https://huggingface.co/).

This means that asking for an uncommon model can make the request _really_ slow. Just loading the model on the GPU already takes a couple dozen seconds, and if the O has to download it as well it could take even minutes. So that's why I opted to limit the models that can be asked here (but there's still a risk no O has it warm and it takes longer than usual).

I believe this whole flow is being improved on by the AI network team, like with the "remote worker" architecture (similar to splitting O+T nodes). Maybe @rickstaa could correct if I said anything wrong or provide more info here 😄